Open Source WG: Specify sources in figures guidelines

AbcSxyZ opened the pull request:
*[In relation to the working group for the 3-7 june book around the question: is The Turing Way Open Source ? (see #3653)]*

fix #3688

Add a section to ask about providing figures sources when adding images around the question of « open source images ».

Remain vague, no guidelines on how/where/whatever (@JimMadge I leave you the fun with the infrastructure WG), in the idea it has to be improved over time.

Use of the qualifier needs, I let you switch to a potential requirement (must) in the future if you consider imposing sources availability. I would love to speak of open source images, I prefer to be patient.

RichardJActon reviewed (commented):
I think that this needs to be expanded on a little to clarify what concretely we want people contributing images to the project to do and why.
There are also some decisions about 'policy' that need to be made by the community about what we ask people to do when they submit an image to the project.

## Figure Sources

Wherever possible include with your image an editable image source or project file for the tool that you used to create the image.
This way things like layers, history, and in purely vector graphics arbitary resolution can be available for these images.
This can make it much easier for someone to pick up where you left off and modify an image for re-use in another context.

For example if an image was created in [inkscape](https://inkscape.org/) the including the `.svg` file, with [krita](https://krita.org/) the `.kra` file or with photoshop `.psd` file.

> I would suggest that it might be a good policy to include the potentially large 'project' files in the the zenodo but not commit these to the git repo to minimise bloat.
> We can continue the practice of commiting the bitmap files to the repo and referencing the DOI of the zenodo repo - ideally the one specific to the version of the image being used.
> This might look something like the below:

Please export the image to a bitmap format such as `.png` or `.jpg` at a suitable resolution and quality for the book and commit this to the git repository.
Add the exported version and the source/project file to the zenodo image repository.

> however there does not really appear to be a standard mechanism for getting images into the 
> create a new zenodo repo in the TTW zenodo community and use this?
> Ask for you image to be added to a new version of an existing zenodo image repo? - Who to ask?

This might be better place in a seperate issue but is related to the process of adding images to the project.

> This might also be an opertunity to think about the alt-text process - should alt text be required before and image is added and how where / should it be stored?
> Can alt text be added to the file metadata to increase potability? - standard here are not yet consistent but see: https://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#alt-text-accessibility

Arielle-Bennett commented:
I agree with @RichardJActon - I also think we may not want to require this, but rather make it clear it's a recommendation to provide source files.

AbcSxyZ commented:
Workflow could be made more obvious, like having a checklist for images, but we discussed it was beyond the scope of this PR.

Following @Arielle-Bennett comment, I mentioned it as a recommendation.

RichardJActon reviewed (changes requested):
A couple of suggested edits

- Minor grammatical changes
- 1 sentence per line
- tweak of clarity on why to include the source in the example

AbcSxyZ commented on book/website/community-handbook/style/style-figures.md:
```diff
@@ -13,6 +13,14 @@ In this sub-chapter we have used the term **figure** to refer to an illustration
+To facilitate figure modification, the availability of images sources is important to enable further editing.
+It is recommended to provide original source files along with the final format (such as `jpg` or `png`) used by figures.
+
+> **Example:** Photoshop source files have a `psd` extension images in this format can be more easily edited before being exported into a `jpg` format.
```
```suggestion
> **Example:** Photoshop uses source files with a `psd` extension to manipulate images layers before an export into a format like `jpg`.
```

AbcSxyZ reviewed (commented):
@RichardJActon Maybe the edit of the example lack a coma to be clear. I suggested some improvements to the overall sentence, let me know what you think of this version (if you want to include the suggestion yourself [not sure if you can]).

Hope it could be understandable somehow for everyone in this way.

**Edit:** I added the modification it to facilitate files review during the day, feel free to suggest improvements.

RichardJActon reviewed (changes requested):
I like this change to include a mention of layers this helps to convey the utility of including the editable format. But I'd suggest this minor tweak.

Sorry for being pedantic about the phrasing here - I don't mean to keep layering you over minutia, It's just because it's very short and we'd like to keep it that way so I'm trying to make implications clear.

JimMadge commented on book/website/community-handbook/style/style-figures.md:
```diff
@@ -13,6 +13,14 @@ In this sub-chapter we have used the term **figure** to refer to an illustration
+To facilitate figure modification, the availability of images sources is important to enable further editing.
+It is recommended to provide original source files along with the final format (such as `jpg` or `png`) used by figures.
+
+> **Example:** Photoshop uses source files with a `psd` extension which allows manipulation of image layers before an export into a format like `jpg`.
```
At the risk of being pedantic, PSD is a file format, usually with a `.psd` extension.
We also might want to use a FOSS and/or open file format example like XCF?

JimMadge reviewed (commented):
I like @RichardJActon and @Arielle-Bennett's comments.

Expanding on _why_ we would ask for this extra effort feels important.

We might also want to be careful about the language. A JPEG _is_ editable. It is a poor choice if you want to continue editing though, as it is a lossy format with (usually) a narrow gamut. Different types of artefacts will get exaggerated with repeat edits.

Maybe there is a hierarchy like,

1. Containers will all data (layers, history, ...), designed for editing (like XCF, Inkscape SVG, PSD)
2. Lossless containers with high colour depth (like TIFF, AVIF)
3. Lossy containers (like JPEG, GIF, AVIF)

I think we can't merge this guidance until we have a clear idea where we are keeping these files too.

RichardJActon commented:
@JimMadge's point about where we want people to put these being something that we should resolve before we merge this guidance is a good one - we probably don't want a bunch big binary files in the git history and as current practice is to add the image files to the git repo we'd be inviting this with the way this is currently phrased.

AbcSxyZ commented:
I tried to improve it to have more careful wording for different states of resources. I added an open source alternative (Inkscape) and fixed files extensions.

> Expanding on why we would ask for this extra effort feels important.

We choosed to emphasis the ability to modify as a reason for sharing sources, you think about a why going beyond that ?

> I think we can't merge this guidance until we have a clear idea where we are keeping these files too.

It leads to the discussion we're having on https://github.com/the-turing-way/the-turing-way/issues/3688 with questions on the entire process for images contribution in the book. It will be important discussions to consider at some points, but which will require specific work by your infrastructure group on how and where to do this.

This addition would be an entry point to bring slowly these questions.

It can let you see what kind of resources you've to deal with when people come up with them, a mean to bring the debate in order to improve process/documentation over the time while structuring management of digital resources and their sources.

I would suggest building on top of it, I doubt you will have soon a clear idea of where these files should be kept.

AbcSxyZ commented:
Asked about psd files briefly:
- The artist save it for ~5 years (personal backup)
- Scriberia organisation will keep it longer. The Turing Way(~'s) PSD files are stored somewhere.

JimMadge commented:
@all-contributors please add @AbcSxyZ for ideas

malvikasharan commented on book/website/community-handbook/style/style-figures.md:
```diff
@@ -32,17 +33,22 @@ In general, make sure to always cite the image properly as directed by the image
-To name your image file, please use all-lowercase and separate words with hyphens (-).
+To facilitate figure modification in the future, it is recommended to share the original/raw or editable version of the image (image sources) when available to go along with the format (such as `jpg` or `png`) used to integrate a figure in the book.
+For example, open source software Inkscape uses `.svg` files and Photoshop uses source files with a `.psd` extension which allows manipulation of image layers before an export into a format like `.jpg`. 
 
```
@RichardJActon Does this address your concern about the location of the raw file?
